sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientOpenReadTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Test;
+using Azure.Storage.Test.Shared;
+using NUnit.Framework;
+
+namespace Azure.Storage.Blobs.Tests
+{
+    [ClientTestFixture(
+        BlobClientOptions.ServiceVersion.V2019_02_02,
+        BlobClientOptions.ServiceVersion.V2019_07_07,
+        BlobClientOptions.ServiceVersion.V2019_12_12,
+        BlobClientOptions.ServiceVersion.V2020_02_10,
+        BlobClientOptions.ServiceVersion.V2020_04_08,
+        BlobClientOptions.ServiceVersion.V2020_06_12,
+        BlobClientOptions.ServiceVersion.V2020_08_04,
+        BlobClientOptions.ServiceVersion.V2020_10_02,
+        BlobClientOptions.ServiceVersion.V2020_12_06,
+        StorageVersionExtensions.LatestVersion,
+        StorageVersionExtensions.MaxVersion,
+        RecordingServiceVersion = BlobClientOptions.ServiceVersion.V2020_12_06,
+        LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion })]
+    public abstract class BlobBaseClientOpenReadTests<TBlobClient> : OpenReadTestBase<
```

Comment:
Could this class extend `BlobTestBase`, so we could avoid the `ClientTestFixture` above?

Reply:
Unfortunately not. It has to extend OpenReadTestBase, as that is where the tests are actually written and need to be inherited from, and OpenReadTestBase has to work on Blobs, DataLake, and Shares, each of which has a different class that defines service versions.

This unfortunately leaves us with the situation where BlobTestBase, BlobBaseClientTransactionalHashingTests, and now BlobBaseClientOpenReadTests must individually set their ClientTestFixture. I see now that this does become annoying to maintain as this scales out and service versions are updated.

Perhaps the fix is to, in each of the service test packages, define a new attribute that extends ClientTestFixture and supplies all these values there. That way there is once again only one place to update service versions in code per-package, it's just no longer on BlobTestBase or equivalent class. Then all these classes can just be tagged with [BlobsTestFixture] or service name equivalent.